packages/api/src/pins.js

```diff
@@ -165,61 +162,46 @@ export async function pinPost (request, env, ctx) {
  * @return {Promise<JSONResponse>}
  */
 async function createPin (pinData, authToken, env, ctx) {
-  const { cid, name, origins, meta } = pinData
-  const normalizedCid = normalizeCid(cid)
-
-  const pinRequestData = {
-    requestedCid: cid,
-    cid: normalizedCid,
-    authKey: authToken
-  }
-  const pinOptions = {}
+  const { cid, origins, meta, normalizedCid } = pinData
 
-  if (name) {
-    pinRequestData.name = name
-    pinOptions.name = name
-  }
+  const pinName = pinData.name || undefined // deal with empty strings
 
-  if (origins) {
-    pinOptions.origins = origins
-  }
+  await env.cluster.pin(cid, {
+    name: pinName,
+    origins,
+    metadata: meta
+  })
+  const pins = await getPins(cid, env.cluster)
 
-  if (meta) {
-    pinOptions.meta = meta
+  const pinRequestData = {
+    sourceCid: cid,
+    contentCid: normalizedCid,
+    authKey: authToken,
+    name: pinName,
+    pins
   }
 
-  // Pin CID to Cluster
-  // TODO: Look into when the returned Promised is resolved to understand if we should be awaiting this call.
-  env.cluster.pin(normalizedCid, pinOptions)
+  const pinRequest = await env.db.createPsaPinRequest(pinRequestData)
 
-  // Create Pin request in db (not creating any content at this stage if it doesn't already exists)
-  const pinRequest = await env.db.createPAPinRequest(pinRequestData)
-
-  /** @type {ServiceApiPinStatus} */
+  /** @type {PsaPinStatusResponse} */
   const pinStatus = getPinStatus(pinRequest)
 
   /** @type {(() => Promise<any>)[]} */
   const tasks = []
 
-  // If we're pinning content that is currently not in the cluster, it might take a while to
-  // get the cid from the network. We check pinning status asyncrounosly.
-  if (pinRequest.pins.length === 0) {
-    tasks.push(async () => {
-      const okPins = await waitToGetOkPins(cid, env.cluster)
-      // Create the content row
-      // TODO: Get dagSize
-      env.db.createContent({ cid: normalizedCid, pins: okPins })
-      for (const pin of okPins) {
-        await env.db.upsertPin(normalizedCid, pin)
-      }
-    })
+  if (!pins.some(p => PIN_OK_STATUS.includes(p.status))) {
+    tasks.push(
+      waitAndUpdateOkPins.bind(
+        null,
+        normalizeCid,
+        env.cluster,
+        env.db)
+    )
   }
 
-  // TODO: Backups. At the moment backups are related to uploads so
+  // TODO(https://github.com/web3-storage/web3.storage/issues/794)
```

Comment:
I think we should remove this todo. This is not where the code for backups of pin requests will live.

Reply:
Ack 👍